### PR TITLE
Crs 584 aggregate sentences

### PR DESCRIPTION
**Pull request by ldlharper**
Changing the way we model consecutive and single term sentences.

This change leaves the Booking.sentences list and the sentences themselves as immutable.
The consecutive sentences are created on the Booking class.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SingleTermSentence.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  fun buildString(): String {
+    return "SingleTermSentence\t:\t\n" +
+      "Number of sentences\t:\t${sentences.size}\n" +
+      "Sentence Types\t:\t$releaseDateTypes\n" +
+      "Number of Days in Sentence\t:\t${getLengthInDays()}\n" +
+      sentenceCalculation.buildString(releaseDateTypes)
+  }
+
+  override fun getLengthInDays(): Int {
+    val firstSentence = sentences.get(0)
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SingleTermSentence.kt:37]: Single terms sentences seem to only be created from two sentences.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/ConsecutiveSentence.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+) : IdentifiableSentence, CalculableSentence, ExtractableSentence {
+  constructor(orderedSentences: List<Sentence>) :
+    this(
+      orderedSentences.minOf(Sentence::sentencedAt),
+      orderedSentences.map(Sentence::offence).minByOrNull(Offence::committedAt)!!,
+      orderedSentences
+    )
+  @JsonIgnore
+  override lateinit var sentenceCalculation: SentenceCalculation
+  @JsonIgnore
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/ConsecutiveSentence.kt:22]: Something with the spacing here?

**Review by joelstobart-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/ConsecutiveSentence.kt (hunk 2 above)
Comment (ldlharper) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/ConsecutiveSentence.kt:25]: Fixed.

**Review by joelstobart-moj — APPROVED** [review]
LGTM

**Pull request merged**